.github/workflows/run-tests.yaml

```diff
@@ -0,0 +1,106 @@
+name: Run Tests
+on:
+  push:
+  workflow_dispatch:
+    inputs:
+      source:
+        description: 'Source of tests'
+        required: true
+        type: string
+        options:
+          - 'specs'
+          - 'sdk'
+        default: 'sdk'
+      specs_url:
+        description: 'Run tests based on specs'
+        required: false
+        type: string
+        default: ''
+      sdk_version:
+        description: 'Run tests based on SDK'
+        required: false
+        type: string
+        default: 'LATEST'
+      sdk_namespace:
+        description: 'SDK to test'
+        required: true
+        type: string
+        options:
+          - rapid
+        default: 'rapid'
+      endpoint_prefix:
+        description: 'Endpoint to prepend specs paths with'
+        required: true
+        type: string
+      jdk:
+        description: 'JDK version to use'
+        required: true
+        type: string
+        options:
+          - '17'
+          - '21'
+        default: '17'
+  workflow_call:
+    inputs:
+      source:
+        description: 'Source of tests'
+        required: true
+        type: string
+        default: 'sdk'
+      specs_url:
+        description: 'Run tests based on specs'
+        required: false
+        type: string
+        default: ''
+      sdk_version:
+        description: 'Run tests based on SDK'
+        required: false
+        type: string
+        default: 'LATEST'
+      sdk_namespace:
+        description: 'SDK to test'
+        required: true
+        type: string
+        default: 'rapid'
+      jdk:
+        description: 'JDK version to use'
+        required: true
+        type: string
+        default: '17'
+      endpoint_prefix:
+        description: 'Endpoint to prepend specs paths with'
+        required: true
+        type: string
+
+jobs:
+  inputs-validation:
+    runs-on: ubuntu-latest
+    steps:
+      - shell: python -u {0}
+        run: |
+          if any([
+            'specs' in '${{ github.event.inputs.source }}' and not('${{ github.event.inputs.specs_url }}'),
+            'sdk' in '${{ github.event.inputs.source }}' and not('${{ github.event.inputs.sdk_version }}')
+          ]):
+            print('::error::Invalid inputs')
```

Comment:
Nice, but you may add a little more details to the message here. Why the input is invalid? Something like:
`Invalid inputs, you need to provide either the specs_url or the sdk_version but not both. You provided specs_url=[...], and sdk_version=[...]`

Reply:
Done.